lib/Sema/LookupVisibleDecls.cpp

```diff
@@ -393,13 +406,17 @@ static DeclVisibilityKind getReasonForSuper(DeclVisibilityKind Reason) {
 static void lookupDeclsFromProtocolsBeingConformedTo(
     Type BaseTy, VisibleDeclConsumer &Consumer, LookupState LS,
     const DeclContext *FromContext, DeclVisibilityKind Reason,
-    VisitedSet &Visited) {
+    VisitedSet &Visited, const bool visitInheritedConformances) {
   NominalTypeDecl *CurrNominal = BaseTy->getAnyNominal();
   if (!CurrNominal)
     return;
   ModuleDecl *Module = FromContext->getParentModule();
 
   for (auto Conformance : CurrNominal->getAllConformances()) {
+    if (!visitInheritedConformances &&
+        Conformance->getKind() == ProtocolConformanceKind::Inherited)
+      continue;
```

Comment:
`visitInheritedConformances` seems to be always `false`. Am I missing something?
Also, could you explain why you added this?

Reply:
> `visitInheritedConformances` seems to be always `false`.

Yes, there is only one call site right now, but I think this behavioural detail is better off being explicit.

> Also, could you explain why you added this?

`lookupDeclsFromProtocolsBeingConformedTo` is called on the class and every superclass in the chain, so inherited conformances are visited at least twice without this flag. Although the repeatedly consumed members are eventually filtered out in `OverrideFilteringConsumer`, we also perform some expensive checks when we iterate over conformances, so I thought this would be a nice optimization.

Reply:
@rintaro I figured we could do better by calling `lookupDeclsFromProtocolsBeingConformedTo` only once, so I factored out this optimization into a separate PR.